domain/storageprovisioning/state/filesystem.go

```diff
@@ -0,0 +1,288 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package state
+
+import (
+	"context"
+	"database/sql"
+	"maps"
+
+	"github.com/canonical/sqlair"
+	"github.com/juju/juju/core/database"
+	"github.com/juju/juju/core/watcher/eventsource"
+	"github.com/juju/juju/domain"
+	"github.com/juju/juju/domain/life"
+	"github.com/juju/juju/domain/storageprovisioning"
+	"github.com/juju/juju/internal/errors"
+)
+
+// GetFilesystemAttachmentIDs returns the
+// [storageprovisioning.FilesystemAttachmentID] information for each filesystem
+// attachment uuid supplied. If a uuid does not exist or isn't attached to
+// either a machine or a unit then it will not exist in the result.
+func (st *State) GetFilesystemAttachmentIDs(
+	ctx context.Context, uuids []string,
+) (map[string]storageprovisioning.FilesystemAttachmentID, error) {
+	db, err := st.DB()
+	if err != nil {
+		return nil, errors.Capture(err)
+	}
+
```

Comment:
Return early if the uuids slice is empty.

Reply:
Done & Done